distributions/R/scripts/new/ctype_translator.py

```diff
@@ -0,0 +1,21 @@
+from ctypes import *
+from collections import defaultdict
+
+
+mapping = defaultdict(lambda: 'error.type', {
+    c_bool: 'logical',
+    c_char: 'character',
+    POINTER(c_char): 'character',
+    c_ubyte: 'raw',
+    c_int16: 'integer',
+    c_int32: 'integer',
+    c_int64: 'integer',
+    c_uint16: 'integer',
+    c_uint32: 'integer',
+    c_uint64: 'integer',
+    c_float: 'single',
+    c_double: 'double'
+})
+
+for ctype in list(mapping.keys()):
```

Comment:
I `list` really needed here  ? `keys()` should return iterable
This loop will also encounter `POINTER(c_char`)` and `POINTER(POINTER(c_char))` will be added as a new key ?

Reply:
This loop modifies `mapping`, so it needs a copy of its keys rather than an iterable.
Adding `POINTER(POINTER(c_char))` is intended, as R can handle an array of strings a.k.a `char **`